RecoTracker/GeometryESProducer/plugins/TrackerRecoGeometryESProducer.cc

```diff
@@ -14,32 +14,47 @@
 #include <memory>
 #include <string>
 
+class TrackerRecoGeometryESProducer : public edm::ESProducer {
+public:
+  TrackerRecoGeometryESProducer(const edm::ParameterSet &p);
+
+  std::unique_ptr<GeometricSearchTracker> produce(const TrackerRecoGeometryRecord &);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+
+private:
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> tTopToken_;
+};
+
 using namespace edm;
 
 TrackerRecoGeometryESProducer::TrackerRecoGeometryESProducer(const edm::ParameterSet &p) {
-  setWhatProduced(this);
+  auto c = setWhatProduced(this);
+
   // 08-Oct-2007 - Patrick Janot
   // Allow several reco geometries to be created, corresponding to the labelled
   // TrackerDigiGeometry's - that must created beforehand. Useful to handle an
   // aligned and a misaligned geometry in the same job.
   // The default parameter ("") makes this change transparent to the user
   // See FastSimulation/Configuration/data/ for examples of cfi's.
-  geoLabel = p.getUntrackedParameter<std::string>("trackerGeometryLabel", "");
+  c.setConsumes(geomToken_, edm::ESInputTag("", p.getUntrackedParameter<std::string>("trackerGeometryLabel")));
+  c.setConsumes(tTopToken_);
 }
 
-TrackerRecoGeometryESProducer::~TrackerRecoGeometryESProducer() {}
-
 std::unique_ptr<GeometricSearchTracker> TrackerRecoGeometryESProducer::produce(
     const TrackerRecoGeometryRecord &iRecord) {
-  edm::ESHandle<TrackerGeometry> tG;
-  iRecord.getRecord<TrackerDigiGeometryRecord>().get(geoLabel, tG);
-
-  edm::ESHandle<TrackerTopology> tTopoHand;
-  iRecord.getRecord<TrackerTopologyRcd>().get(tTopoHand);
-  const TrackerTopology *tTopo = tTopoHand.product();
+  TrackerGeometry const &tG = iRecord.get(geomToken_);
 
   GeometricSearchTrackerBuilder builder;
-  return std::unique_ptr<GeometricSearchTracker>(builder.build(tG->trackerDet(), &(*tG), tTopo));
+  return std::unique_ptr<GeometricSearchTracker>(builder.build(tG.trackerDet(), &tG, &iRecord.get(tTopToken_)));
+}
+
+void TrackerRecoGeometryESProducer::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.addUntracked<std::string>("trackerGeometryLabel", "");
+  descriptions.addDefault(desc);
```

Comment:
@perrotta I do not believe this is the same case as in #27589. Here I just changed accessing an untracked parameter with a default to injecting the default via a `fillDescriptions`. To the framework and any outside observer (or even this module) the two ways of doing it are identical. The only difference is now `edmPluginHelp` will actually work.